Project-security/csrf/my-site/routes/demo.js

```diff
@@ -0,0 +1,187 @@
+const express = require('express');
+const bcrypt = require('bcryptjs');
+
+const db = require('../data/database');
+
+const router = express.Router();
+
+router.get('/', function (req, res) {
+  res.render('welcome');
+});
+
+router.get('/signup', function (req, res) {
+  let sessionInputData = req.session.inputData;
+
+  if (!sessionInputData) {
+    sessionInputData = {
+      hasError: false,
+      email: '',
+      confirmEmail: '',
+      password: '',
+    };
+  }
+
+  req.session.inputData = null;
+
+  res.render('signup', { inputData: sessionInputData });
+});
+
+router.get('/login', function (req, res) {
+  let sessionInputData = req.session.inputData;
+
+  if (!sessionInputData) {
+    sessionInputData = {
+      hasError: false,
+      email: '',
+      password: '',
+    };
+  }
+
+  req.session.inputData = null;
+  res.render('login', { inputData: sessionInputData });
+});
+
+router.post('/signup', async function (req, res) {
+  const userData = req.body;
+  const enteredEmail = userData.email; // userData['email']
+  const enteredConfirmEmail = userData['confirm-email'];
+  const enteredPassword = userData.password;
+
+  if (
+    !enteredEmail ||
```

Comment:
again multiline conditions ...
remember to alway use `||` etc at the begining of line (from second line ofcourse)

Reply:
we will need to think about `lint` later on in this case

Reply:
Well it was project included in a course that show what this kind of security problems there are. So basically it was 3 mini projects in 1 large project.